server/webapp/WEB-INF/rails.new/webpack/views/dashboard/personalization_modal_widget.js.msx

```diff
@@ -121,6 +123,17 @@ const AlertContainer = {
   }
 };
 
+const ToolTip = {
+  view(vnode) {
+    return <span class="tooltip-outer-container">
+      <span class="tooltip-inner-container">
+        <i class="tooltip-hint" onclick={ (e) => { e.stopPropogation; e.target.nextSibling.classList.toggle("show"); }}/>
```

Comment:
It's usually considered a risky thing to manipulate DOM outside of mithril, not to mention causes repaints outside of the mithril framework.

The right way to do this would be to remember the "toggle state" in the controller:

```
const Tooltip = {
  oninit(vnode) {
    vnode.state.showTooltip = Stream(false)
    vnode.state.toggleTooltip = function() {vnode.state.showTooltip(!(vnode.state.showTooltip))}
  },
  view(vnode) {
    <i class='...' onclick={vnode.state.toggleTooltip.bind(vnode.state)} />
    <span class=`tooltip ${vnode.state.showTooltip() ? 'show' : 'hide'}`>...</span>
  }
}
```

Reply:
Oddly, this is how the tooltips in `buttonWithTooltip` and `linkWithTooltip` work in `form_helper`. But I do agree as a general guideline that mithril likes to manage this state.

Reply:
But more importantly, this implementation for `onclick` means that it's possible to open more than one tooltip, which is strange. Also, clicking outside will not clear the tooltip. Clicking the link needs to clear the tolip as well. `:hover` is much simpler, but I get why you needed `onclick`.